### DISCUSSION TITLE
Submit to Coolify

### DISCUSSION THREAD

**Discussion by AlejandroAkbal** [maelgangloff/domain-watchdog#96] [General]
Coolify has 60k users that could easily enjoy a one click install of this wonderful project

![image](https://github.com/user-attachments/assets/a703086f-3185-455b-be88-97ce9d1af31b)

You can create a template for them to install with one click following this docs: https://coolify.io/docs/knowledge-base/add-a-service

**Comment by maelgangloff** [comment]
Thanks but it would require too much code restructuring to support these new environment variables and would require adding more configuration.
I'll leave this issue open if anyone wants to do it.

**Comment by AlejandroAkbal** [comment]
There is not new environment variables, Coolify does not require anything special, its just docker compose files, special env vars are a plus for automatic configuration, but are not required

See: https://github.com/coollabsio/coolify/blob/main/templates/compose/n8n.yaml

**Comment by AlejandroAkbal** [comment]
WIP version:

```sh
# documentation: https://github.com/maelgangloff/domain-watchdog
# slogan: Monitor's domains and purchases them when available.
# tags: domain,watchdog,monitoring,management
# logo: svgs/domain-watchdog.svg
# port: 8080

services:
    domainwatchdog:
        image: maelgangloff/domain-watchdog:latest
        environment:
            SERVER_NAME: http://${SERVER_NAME:-localhost}
            DATABASE_URL: postgresql://${SERVICE_USER_POSTGRES}:${SERVICE_PASSWORD_POSTGRES}@database:5432/domainwatchdog?serverVersion=${POSTGRES_VERSION:-16}&charset=utf8
            APP_SECRET: ${SERVICE_BASE64_DOMAINWATCHDOG}
            REGISTRATION_ENABLED: ${REGISTRATION_ENABLED:-true}
            REGISTRATION_VERIFY_EMAIL: ${REGISTRATION_VERIFY_EMAIL:-false}
            LIMITED_FEATURES: ${LIMITED_FEATURES:-false}
            LIMIT_MAX_WATCHLIST: ${LIMIT_MAX_WATCHLIST:-0}
            LIMIT_MAX_WATCHLIST_DOMAINS: ${LIMIT_MAX_WATCHLIST_DOMAINS:-0}
            LIMIT_MAX_WATCHLIST_WEBHOOKS: ${LIMIT_MAX_WATCHLIST_WEBHOOKS:-0}
            MAILER_DSN: ${MAILER_DSN:-null://null}
        volumes:
            - caddy_data:/data
            - caddy_config:/config
            - domainwatchdog_content:/app/public/content

    php-worker:
        image: maelgangloff/domain-watchdog:latest
        restart: always
        command: php /app/bin/console messenger:consume --all --time-limit=3600 -vvv
        environment:
            DATABASE_URL: postgresql://${SERVICE_USER_POSTGRES}:${SERVICE_PASSWORD_POSTGRES}@database:5432/domainwatchdog?serverVersion=${POSTGRES_VERSION:-16}&charset=utf8
            APP_SECRET: ${SERVICE_BASE64_DOMAINWATCHDOG}
            MAILER_DSN: ${MAILER_DSN:-null://null}
        healthcheck:
            disable: true

    database:
        image: postgres:${POSTGRES_VERSION:-16}-alpine
        environment:
            POSTGRES_DB: domainwatchdog
            POSTGRES_PASSWORD: ${SERVICE_PASSWORD_POSTGRES}
            POSTGRES_USER: ${SERVICE_USER_POSTGRES}
        healthcheck:
            test: [ "CMD", "pg_isready", "-d", "domainwatchdog", "-U", "${SERVICE_USER_POSTGRES}" ]
            timeout: 5s
            retries: 5
            start_period: 60s
        volumes:
            - database_data:/var/lib/postgresql/data:rw
            
```

**Reply by Recreabox** [reply]
> WIP version:
> 
> ```shell
> # documentation: https://github.com/maelgangloff/domain-watchdog
> # slogan: Monitor's domains and purchases them when available.
> # tags: domain,watchdog,monitoring,management
> # logo: svgs/domain-watchdog.svg
> # port: 8080
> 
> services:
>     domainwatchdog:
>         image: maelgangloff/domain-watchdog:latest
>         environment:
>             SERVER_NAME: http://${SERVER_NAME:-localhost}
>             DATABASE_URL: postgresql://${SERVICE_USER_POSTGRES}:${SERVICE_PASSWORD_POSTGRES}@database:5432/domainwatchdog?serverVersion=${POSTGRES_VERSION:-16}&charset=utf8
>             APP_SECRET: ${SERVICE_BASE64_DOMAINWATCHDOG}
>             REGISTRATION_ENABLED: ${REGISTRATION_ENABLED:-true}
>             REGISTRATION_VERIFY_EMAIL: ${REGISTRATION_VERIFY_EMAIL:-false}
>             LIMITED_FEATURES: ${LIMITED_FEATURES:-false}
>             LIMIT_MAX_WATCHLIST: ${LIMIT_MAX_WATCHLIST:-0}
>             LIMIT_MAX_WATCHLIST_DOMAINS: ${LIMIT_MAX_WATCHLIST_DOMAINS:-0}
>             LIMIT_MAX_WATCHLIST_WEBHOOKS: ${LIMIT_MAX_WATCHLIST_WEBHOOKS:-0}
>             MAILER_DSN: ${MAILER_DSN:-null://null}
>         volumes:
>             - caddy_data:/data
>             - caddy_config:/config
>             - domainwatchdog_content:/app/public/content
> 
>     php-worker:
>         image: maelgangloff/domain-watchdog:latest
>         restart: always
>         command: php /app/bin/console messenger:consume --all --time-limit=3600 -vvv
>         environment:
>             DATABASE_URL: postgresql://${SERVICE_USER_POSTGRES}:${SERVICE_PASSWORD_POSTGRES}@database:5432/domainwatchdog?serverVersion=${POSTGRES_VERSION:-16}&charset=utf8
>             APP_SECRET: ${SERVICE_BASE64_DOMAINWATCHDOG}
>             MAILER_DSN: ${MAILER_DSN:-null://null}
>         healthcheck:
>             disable: true
> 
>     database:
>         image: postgres:${POSTGRES_VERSION:-16}-alpine
>         environment:
>             POSTGRES_DB: domainwatchdog
>             POSTGRES_PASSWORD: ${SERVICE_PASSWORD_POSTGRES}
>             POSTGRES_USER: ${SERVICE_USER_POSTGRES}
>         healthcheck:
>             test: [ "CMD", "pg_isready", "-d", "domainwatchdog", "-U", "${SERVICE_USER_POSTGRES}" ]
>             timeout: 5s
>             retries: 5
>             start_period: 60s
>         volumes:
>             - database_data:/var/lib/postgresql/data:rw
>             
> ```

Hello !

Just here to say that i am also searching for a docker-compose template that can work with Coolify
This one, doesn't works

I confirm, there is no need to do a specific development

**Comment by shamshuddinlakhani-cloud** [comment]
You can submit this project to Coolify by creating a custom service template using their docs, linking your Docker setup and defining the volumes, ports, and environment variables. Once added, their 60k+ users could enjoy a one-click install with persistent storage and proper configuration for your app.
